File: javascript/58_Length_of_Last_Word.js
/**
 * @param {string} s
 * @return {number}
 */
 var lengthOfLastWord = function(s) {
    let cnt =  0; 
    let i = s.length-1;

    while (s[i] == ' ') i--;
    while (i >= 0 && s[i] != ' ') {
        cnt ++;
        i--;
    }
    return cnt;
};

 var lengthOfLastWord2 = function(s) {
    let cnt = 0;
    for (let i = s.length-1; i >= 0; i--) {
        if (cnt != 0 && s[i] == " ") {
            return cnt;
        }
        if (s[i] != " ") {
            cnt ++;
        }
    }
    return cnt;
};

output1 = lengthOfLastWord("Hello World");
output2 = lengthOfLastWord("   fly me   to   the moon  ");
output3 = lengthOfLastWord("luffy is still joyboy");
console.log(output1, output2, output3);

File: python/58_Length_of_Last_Word.py
'''
Time: O(n)
Space: O(1)
'''

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i, cnt = len(s)-1, 0
        while s[i] == ' ':
            i -= 1
        while i >= 0 and s[i] != ' ':
            cnt += 1
            i -= 1
        
        return cnt


output1 = Solution().lengthOfLastWord("Hello World")
output2 = Solution().lengthOfLastWord("   fly me   to   the moon  ")
output3 = Solution().lengthOfLastWord("luffy is still joyboy")
print(output1, output2, output3)